Add basic issue templates

robguy21 opened the pull request:
This should lay the ground work for future iterations on templates for bug reports and feature requests

robguy21 committed:
Add basic issue templates

This should lay the ground work for future iterations on templates for bug reports and feature requests

robguy21 committed:
Added priorities to the bug report template

countnoobula commented on .github/ISSUE_TEMPLATE/feature_request.md:
```diff
@@ -0,0 +1,20 @@
+---
+
+**Is your feature request related to a problem? Please describe.**
+A clear and concise description of what the problem is. Ex. I'm always frustrated when [...]
```
I think these should be _italics_ like we have them on Allianz, as this differentiates between the user input and template design.

That would mean it would look like:

--- 

**Is your feature request related to a problem? Please describe.**
_A clear and concise description of what the problem is. Ex. I'm always frustrated when [...]_

adameier commented on .github/ISSUE_TEMPLATE/bug_report.md:
```diff
@@ -0,0 +1,39 @@
+-------------------
+Please tick the relevant checkbox 
+- [ ] Urgent - _The bug is breaking interactions for users_
+- [ ] Critical - _This impacts a critical procedure_
```
Does an issue have to be only one of these? because then I'm not sure what the distinction between Urgent and Critical is

countnoobula replied:
I think Urgent is user visible, where as critical is a background process.

Urgent - It interrupts the user flow.
Critical - It interrupts sync or background tasks (such as sending mails, or syncing data?).

@robguy21 can you confirm?

robguy21 replied:
Correct, I wanted to differentiate between "this is breaking for users" and "this is breaking non-user-facing functionality".

For example - the queues not being able to run versus "add a payment" resulting in broken page [luxity]

sscholle replied:
`critical procedure` is somewhat vague and can be confused, as user interactions can be critical procedures too....

adameier replied:
perhaps a distinction between a bug being user-facing or not, as well as it being critical or not. And a combination of the two can be selected.
e.g. a bug can be user-facing and be either critical or not, the same goes for a bug that isnt user-facing.

sscholle commented on .github/ISSUE_TEMPLATE/bug_report.md:
```diff
@@ -0,0 +1,39 @@
+- [ ] Important - _This should be resolved within a week_
+- [ ] Backlog - _tackle important issues before attending to this this_
+
+Any other comments? (logs, output, error reports, or any other useful information)
```
`Extra Notes` - i think is a clear enough description, no need for a long list of examples

pruhesi commented on .github/ISSUE_TEMPLATE/bug_report.md:
```diff
@@ -0,0 +1,39 @@
+
+---
+
+(Thanks for reporting an issue! Please make sure you click the link above to view the issue guidelines, then fill out the blanks below.)
```
link above?

pruhesi commented:
@robguy21 @countnoobula Just an inquiry about whether issues created by devs would need to follow the above format?

robguy21 commented:
@pruhesi yes- if a developer is making a feature request or a bug report

robguy21 committed:
Rework Priorities and Notes

sscholle commented on .github/ISSUE_TEMPLATE/bug_report.md:
```diff
@@ -0,0 +1,37 @@
+- [ ] Extreme - _The bug is breaking interactions for users, meaning that they are unable to use the system_
+- [ ] Urgent - _This impacts a system procedure in a critical manner_
+- [ ] High - _This should be resolved within a week_
+- [ ] Normal - _tackle important issues before attending to this this_
```
this this

sscholle commented on .github/ISSUE_TEMPLATE/bug_report.md:
```diff
@@ -0,0 +1,37 @@
+Priority
+-------------------
+Please tick the relevant checkbox
+- [ ] Extreme - _The bug is breaking interactions for users, meaning that they are unable to use the system_
```
extreme and urgent seem to be the same - maybe have this level take high priority meaning it goes to top of todo list. - something that is critical goes here
High - means what it says - good - usually a style/UX/ or bug fix that can have impact in the experience
Normal - add it to the bottom - usually a minor tweak and new feature requests

sscholle commented on .github/ISSUE_TEMPLATE/feature_request.md:
```diff
@@ -0,0 +1,20 @@
+_A clear and concise description of what you want to happen._
+
+**Describe alternatives you've considered**
+_A clear and concise description of any alternative solutions or features you've considered._
```
i'd imagine `clear and concise` is assumed as we have so many headings to fill information under - so probably no need for that much repetition

robguy21 committed:
Updated bug report categories to be more clear

This removes the abstraction of "extreme/urgent/high/normal" and prefers
to just be direct about what the category represents

robguy21 committed:
Changed format of feature request to be more "story" based

sscholle commented on .github/ISSUE_TEMPLATE/bug_report.md:
```diff
@@ -0,0 +1,37 @@
+Please tick the relevant checkbox
+- [ ] Prevents the user completing work flows - _The bug is breaking interactions for users, meaning that they are unable to use the system in an expected way_
+- [ ] Prevents the system from completing expected processes - _This bug prevents the system from completing an expected task_
+- [ ] Impacts the system critically - _The system is unable to perform tasks, but these tasks are not critical_
```
confusing: `critically` and `not critical` - which one is it?

robguy21 committed:
Removed confusing critcal references

willemwe0 commented on .github/ISSUE_TEMPLATE/feature_request.md:
```diff
@@ -0,0 +1,20 @@
+
+---
+
+## Description of Existing Problem
```
A good requirement or user story describes the problem, not a solution.

https://medium.com/101ideasforagileteams/problem-centric-user-stories-9815ba23e132

Therefore I think we can leave this out and make the bottom one:
> User Story describing the problem.

Also since user stories are new, lets leave link for how to write them? Like we do with Loom

robguy21 commented:
```
---
name: Feature request
about: Suggest an idea for this project
title: "[FEATURE REQUEST] "
labels: ''
assignees: ''

---

## Description of Existing Problem
_What problem has led to the requirement of this feature request_

## User Story describing required solution
_A user story describing the feature to be implemented_

## Technical Description (not required initially)
_A technical description of the problem, this should be completed by the engineering manager or product manager_

## Additional Resources
_Add any context, screenshots, designs, links, etc. about the feature request here._

# VERSUS


## Problem Centric User Story
_A user story describing the feature to be implemented and the problem_

## Acceptance Criteria
_requirements for featured to be considered implemented_

## Technical Description (not required initially)
_A technical description of the problem, this should be completed by the engineering manager or product manager_

## Additional Resources
_Add any context, screenshots, designs, links, etc. about the feature request here._
```

pruhesi commented:
My vote is yes to the first one and yes to add acceptance criteria to the first one

countnoobula commented:
### User Story
**As a**: [Role]

**I want to**: [Brief description of feature from user perspective]

**Because**: [Brief reason for request of feature]

### Acceptance Criteria
_List of conditions that mark the story as complete_
- [ ] ...

### Edge cases
_List of edge cases to be considered when marking the story as complete_
- [ ] ...

robguy21 committed:
Modified feature request template as per community vote
